src/features/generic/time-utils.ts

```diff
@@ -56,6 +56,31 @@ namespace TimeUtilities {
       displayInternal: displayInternal,
     }
   }
-};
+
+  export function calculateWorkedTimeAndBillableHours(user: TimeEntryTotalDto | DailyCombinedData){
+
+    // calculates billable hours and adds percentage sign
+    const billableHours = (user.projectTime/user.expected * 100).toFixed(1); 
+    const billableHoursWithPercentage = billableHours.concat("%");
+
+    //turns undertime hours positive and calculates hours and minutes
+    const undertimeHours = Math.floor(user.total * -1 / 60)
+    const undertimeMinutes = user.total * -1 % 60;
+
+    //calculates overtime hours and minutes
+    const overtimeHours = Math.floor(user.total / 60)
+    const overtimeMinutes = user.total % 60;
+
+    const undertimeMessage = "Undertime: " + undertimeHours.toString() + "h " + undertimeMinutes.toString() + " minutes";
+    const overtimeMessage = "Overtime: " + overtimeHours.toString() + "h " + overtimeMinutes.toString() + " minutes";
+
+    return {
+      undertimeMessage: undertimeMessage,
+      overtimeMessage: overtimeMessage,
```

Comment:
These can be combined (check previous comments)

Reply:
Done